### PR TITLE
fix(LSP): Apply changes from the language server that touches the most files

### PR DESCRIPTION
**Pull request by peterfication**
Fixes https://github.com/johmsalas/text-case.nvim/issues/107

### CODE REVIEW

**Comment by peterfication** [comment]
@johmsalas unfortunately, I can't test it manually, because I can't reproduce it. And the test is just a unit test, not an integration test.

File: tests/textcase/lsp/error_handling_no_lsp_spec.lua [hunk 1]
```diff
@@ -119,4 +121,103 @@ describe("LSP renaming", function()
...
+    after_each(function()
+      vim.lsp.get_active_clients = get_active_clients_fn
+      vim.lsp.buf_request_all = buf_request_all_fn
+      vim.lsp.get_client_by_id = get_client_by_id_fn
+      vim.lsp.util.make_position_params = make_position_params_fn
+      vim.lsp.util.apply_workspace_edit = apply_workspace_edit_fn
+      vim.api.nvim_err_writeln = err_fn
+    end)
+
+    it("should use the results from the language server that touches the most files", function()
```
[the remark this answers is not in the document]
Comment (peterfication) [reply tests/textcase/lsp/error_handling_no_lsp_spec.lua]: Good idea. Done :)

**Comment by johmsalas** [comment]
Manual Testing

**On main:**
https://github.com/johmsalas/text-case.nvim/assets/6575405/65b71e18-867c-4ff7-8a76-6f5cdff610ab

**On feat-lsp-apply-most-changes:**
https://github.com/johmsalas/text-case.nvim/assets/6575405/ba713372-f883-40ff-9aee-252109dff040

File: lua/textcase/plugin/conversion.lua [hunk 2]
```diff
@@ -96,17 +96,31 @@ function M.do_lsp_rename(method)
...
         end
       end
+
+      if total_files > 0 and offset_encoding_to_be_applied ~= nil then
+        -- If there are results to be applied, apply them.
+        vim.lsp.util.apply_workspace_edit(results_to_be_applied, offset_encoding_to_be_applied)
+      end
+
+      -- After the edits are applied, the files are not saved automatically.
+      -- Let's remind ourselves to save those...
```
Comment (johmsalas) [inline lua/textcase/plugin/conversion.lua:123]: xD thanks

**Review by johmsalas — APPROVED** [review]
Your code is pretty clear and well named
I think it is a fix, not a feat. 

Thank you so much for fixing it, it was affecting me :)

**Pull request merged**